Sources/Sentry/SentryStacktraceBuilder.m

```diff
@@ -56,6 +59,41 @@ - (SentryStacktrace *)retrieveStacktraceFromCursor:(SentryCrashStackCursor)stack
     return stacktrace;
 }
 
+- (SentryStacktrace *)retrieveStacktraceFromCursor:(SentryCrashStackCursor)stackCursor
+{
+    return [self retrieveStackTraceFromCursor:&stackCursor
+                        withFrameSymbolicator:^SentryFrame *(SentryCrashStackCursor *cursor) {
+                            if (stackCursor.symbolicate(cursor)) {
+                                return [self.crashStackEntryMapper mapStackEntryWithCursor:*cursor];
+                            }
+                            return nil;
+                        }];
+}
+
+- (SentryStacktrace *)retrieveStacktraceFromCursorNatively:(SentryCrashStackCursor)stackCursor
+{
+    /**
+     * Different from `retrieveStacktraceFromCursor`, this method uses `dladdr` to retrieve
+     * information from the stack cursor, which is much faster and thread safe but is not async
+     * safe, that's why this method cannot be used during crashes.
+     */
+
+    return [self retrieveStackTraceFromCursor:&stackCursor
+                        withFrameSymbolicator:^SentryFrame *(SentryCrashStackCursor *cursor) {
+                            Dl_info info = { 0 };
+                            if (dladdr((const void *)cursor->stackEntry.address, &info)) {
```

Comment:
`h`: `sentrycrashdl_dladdr` uses `CALL_INSTRUCTION_FROM_RETURN_ADDRESS`. Why don't we use it here?

Reply:
@brustolin, please have a look at this vital comment.

Reply:
Sorry, somehow I miss this.